### PR TITLE
Add rich nav artifact upload param

### PR DESCRIPTION
**Pull request by brianaeng**
Adding a new var to map to the uploadRichNavArtifacts parameter, need this base change to use the var in the runtime repo's build yml.

### CODE REVIEW

**Comment by brianaeng** [comment]
@safern FYI here's a starter PR for adding the param we discussed via email! @jepetty is doing an experiment w/ env vars to see if we can get away with not needing this change though, so please hold on merging 😄

File: eng/common/templates/job/job.yml [hunk 1]
```diff
@@ -133,6 +133,7 @@ jobs:
       inputs:
         languages: ${{ coalesce(parameters.richCodeNavigationLanguage, 'csharp') }}
         environment: ${{ coalesce(parameters.richCodeNavigationEnvironment, 'production') }}
+        uploadRichNavArtifacts: ${{ coalesce(parameters.richCodeNavigationArtifactsUpload, 'false') }}
```
Comment (safern) [inline eng/common/templates/job/job.yml:136]: For this to be set to true, we would need to pass this parameter down as true at the current callsites no?

Comment (brianaeng) [reply eng/common/templates/job/job.yml:136]: Not sure if this answers your question (let me know if it doesn't!) but yeah we'll need to set to true in the runtime richnav yml. 

Here's that part that I haven't PR'd into runtime yet (holding on Jess' investigation and/or this PR): https://github.com/brianaeng/runtime/pull/1/files

Comment (safern) [reply eng/common/templates/job/job.yml:136]: Sounds good.

**Comment by brianaeng** [comment]
@jepetty got the env variable approach working to enable the richnav artifacts upload, so we don't need this PR anymore (fyi @safern -- we'll do a PR to the runtime repo soon).

**Pull request closed without merging**